### PR TITLE
[Bugfix:GradeInquiry] Defined and implemented Grade Inquiry buttons

### LINKED ISSUE #3921: Tweaks to grade inquiry UI
change first (white) 'submit grade inquiry' button to be 'open grade inquiry'
leave text on (blue) 'submit grade inquiry' button

when a student views a resolved grade inquiry, should have 3 options (approximate text):
(blue) reopen grade inquiry with response
(white) reopen grade inquiry with no response
(white) post message without reopening inquiry
--  maybe 2 of those options are disabled if the textbox is currently empty???  (forcing you to write a message)
-- other buttons should similarly be disabled if the textbox is empty (e.g., submitting the first post in a grade inquiry)

From the TA interface, for a resolved grade inquiry, change the "Submit Grade Inquiry" button to instead say "Reopen Grade Inquiry"

ALSO...  I left a bunch of "TODO"s in the student/grader/instructor documentation pages (I'll make & merge that PR very soon)

### LINKED ISSUE #4084: Limited access grader can see all student's grade inquiries
**Describe the bug**
If limited access graders can url hack themselves to a student unassigned to them they can see the student's grade inquiry and posts

**Expected behavior**
There should be a check to see if the user can view the grade inquiry discussion from the panel (is the user a full access grader or a limited access grader assigned to the student)

### PR DESCRIPTION
**Pull request by sak6lab**
### What is the current behavior?
Buttons in both grading view and student's submission view is unclear.
closes #3921 

### What is the new behavior?
- Buttons appear as follows
grade-inquiry-breakdown
- If there is no submission for the grader the text area and buttons are not shown rather than a disabled button
grader_no_submission
 
Requested changes from @jmcknight98:
closes #4084 and #4085

### CODE REVIEW

File: site/app/templates/submission/regrade/Discussion.twig [hunk 1]
```diff
@@ -79,60 +73,82 @@
...
-            {% endif %}
-        </div>
-    </form>
+    {% if grade_inquiry_status != 'no_submission' %}
+        <form method="POST" id="reply-text-form" >
+            <input type="hidden" id="thread_id" name="thread_id" value="{{ thread_id }}">
+            <input type="hidden" id="gradeable_id" name="gradeable_id" value="{{ gradeable_id }}">
+            <input type="hidden" id="submitter_id" name="submitter_id" value="{{ submitter_id }}">
+
+                <label for="reply-text-area">Write a inquiry:</label>
```
Comment (jmcknight98) [inline site/app/templates/submission/regrade/Discussion.twig]: Should maybe be changed to "an inquiry" or "a grade inquiry"

**Comment by jmcknight98** [comment]
It might be nice if the "Write a inquiry" display message also changed depending on whether an inquiry is in opened/resolved. 

Also the "Warning: Frivolous grade inquiries may lead to grade deductions or lost late days" message appears for instructors/graders, which I think is kind of strange. 

These are just suggestions that can be ignored, I don't know if would be worth changing or even should be changed, just my thoughts when navigating through the pages

**Comment by sak6lab** [comment]
- Limited Access Graders are supposed to be able to interact with grade inquiries but I could not find an easy way to check for that so I made an issue #4085 and im still working on it. 
- The "Warning: Frivolous grade inquiries may lead to grade deductions or lost late days" is message set at setup of a submitty course so I just left it. It is message set by the instructor to be displayed before any grade inquiry.

**Comment by sak6lab** [comment]
fixed #4084 and #4085

**Comment by jmcknight98** [comment]
The Limited Access Graders do work now.  There is a new error though when trying to go to view grade as a student (looks like there is probably just a missing can_inquiry somewhere):
aaaaError

**Pull request merged**